File: backend/models/itemModel.js
const mongoose = require('mongoose');
const { Schema } = mongoose;

const itemSchema = new Schema({
  itemId: {
    type: Schema.Types.ObjectId,
    required: true,
    unique: true
  },
  itemFunc: {
    type: String,
    required: true
  },
  itemName: {
    type: String,
    required: true
  },
  itemType: {
    type: Number,
    enum: [0, 1], // 0:素材, 1:消耗物
    required: true
  },
  itemEffect: {
    type: String,
    required: true
  },
  maxStack: {
    type: Number,
    default: 1
  },
  itemMethod: {
    type: String,
    enum: ['npc', 'chest', 'scan', 'randomEvent'],
    required: true
  },
  itemRarity: {
    type: Number,
    min: 1,
    max: 5,
    required: true
  },
  resultId: {
    type: Schema.Types.ObjectId,
    default: null
  }
});

// 明確指定集合名稱為 'items'
const Item = mongoose.model('Item', itemSchema, 'items');

module.exports = Item;

File: frontend/app/src/main/java/com/ntou01157/hunter/models/model_api/ItemModel.kt
// package com.ntou01157.hunter.Backpack.model
package com.ntou01157.hunter.model.model_api

import com.google.gson.annotations.SerializedName

data class Item(
    @SerializedName("_id") val itemId: String,
    @SerializedName("itemFunc") val itemFunc: String,
    @SerializedName("itemName") val itemName: String,
    @SerializedName("itemType") val itemType: Int,
    @SerializedName("itemEffect") val itemEffect: String,
    @SerializedName("itemMethod") val itemMethod: String,
    @SerializedName("itemRarity") val itemRarity: Int,
    @SerializedName("resultId") val resultId: String? = null,
    @SerializedName("itemPic") val itemPic: String = "default_itempic" // 預設圖片名稱
)
